deeplearning4j-scaleout/spark/dl4j-spark-ml/src/main/scala/org/deeplearning4j/spark/ml/impl/SparkDl4jNetwork.scala

```diff
@@ -0,0 +1,125 @@
+import org.deeplearning4j.nn.multilayer.MultiLayerNetwork
+import org.deeplearning4j.optimize.api.IterationListener
+import org.deeplearning4j.spark.impl.multilayer.SparkDl4jMultiLayer
+import org.deeplearning4j.spark.impl.paramavg.ParameterAveragingTrainingMaster
+import org.deeplearning4j.util.ModelSerializer
+import org.nd4j.linalg.dataset.DataSet
+import org.nd4j.linalg.factory.Nd4j
+import org.nd4j.linalg.util.FeatureUtil
+
+
+final class SparkDl4jNetwork(override val uid: String) extends Predictor[Vector, SparkDl4jNetwork, SparkDl4jModel]  {
+
+    private var _multiLayerConfiguration : MultiLayerConfiguration = _
+    private var _numLabels : Int = 2
+    private var _freq : Int = 10
+    private var _trainingMaster : Serializer = _
+    private var _listeners : util.Collection[IterationListener] = _
+    private var _collectStats: Boolean = false
+
+    def this() = this(Identifiable.randomUID("dl4j"))
+
+    override def train(dataset: DataFrame) : SparkDl4jModel = {
+        val sparkNet = new SparkDl4jMultiLayer(dataset.sqlContext.sparkContext, _multiLayerConfiguration, _trainingMaster())
+        sparkNet.setCollectTrainingStats(_collectStats)
+        if (_listeners != null) {
+            sparkNet.setListeners(_listeners)
+        }
+        val lps = dataset.select(getFeaturesCol, getLabelCol).rdd
+            .map(row => new LabeledPoint(row.getAs[Double](getLabelCol), row.getAs[Vector](getFeaturesCol)))
+            .map(item => {
+                val features = item.features
+                val label = item.label
+                if (_numLabels > 1) {
+                    new DataSet(Nd4j.create(features.toArray), FeatureUtil.toOutcomeVector(label.asInstanceOf[Int], _numLabels))
+                } else {
+                    new DataSet(Nd4j.create(features.toArray), Nd4j.create(Array(label)))
+                }
+            })
+        sparkNet.fit(lps)
+        new SparkDl4jModel(uid, sparkNet)
+    }
+
+    override def copy(extra: ParamMap) : SparkDl4jNetwork = defaultCopy(extra)
+
+    def setNumLabels(value: Int) : SparkDl4jNetwork = {
+        this._numLabels = value
+        this
+    }
+
+    def setMultiLayerConfiguration(multiLayerConfiguration: MultiLayerConfiguration) : SparkDl4jNetwork = {
+        this._multiLayerConfiguration = multiLayerConfiguration
+        this
+    }
+
+    def setTrainingMaster(tm: Serializer) : SparkDl4jNetwork = {
+        this._trainingMaster = tm
+        this
+    }
+
+    def setListeners(iterationListener: util.Collection[IterationListener]) : SparkDl4jNetwork = {
+        this._listeners = iterationListener
+        this
+    }
+
+    def setCollectionStats(collectStats: Boolean) : SparkDl4jNetwork = {
+        this._collectStats = collectStats
+        this
+    }
+}
+
+class SparkDl4jModel(override val uid: String, network: SparkDl4jMultiLayer)
+    extends PredictionModel[Vector, SparkDl4jModel] with Serializable with MLWritable {
```

Comment:
I don't know much about dl4j's computation graphs. I am looking at adding an autoencoder, word2vec, and paragraph vectors as a transformer (where you can apply the transformations to features through a ml pipeline). Maybe computation graphs could follow that pattern. It is slightly different than spark's predictor apis, and I think it can handle more data types.